File: leetcode/344.reverse-string/ts/reversestring.test.ts
import { test, expect } from "vitest"
import reverseString from "./reversestring.js"

interface testcase {
    s: string[]
    want: string[]
}

test("reversestring", () => {
    const tcs: testcase[] = [
        { s: ["h", "e", "l", "l", "o"], want: ["o", "l", "l", "e", "h"] },
        { s: ["H", "a", "n", "n", "a", "h"], want: ["h", "a", "n", "n", "a", "H"] },
    ]

    tcs.forEach(tc => {
        reverseString(tc.s)
        expect(tc.want === tc.s)
    })
})

File: leetcode/344.reverse-string/cpp/reversestring_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "reversestring.cpp"

struct testCase
{
    std::vector<char> s;
    std::vector<char> want;
};

TEST_CASE("ReverseString")
{
    std::vector<testCase> tcs{
        {{'h', 'e', 'l', 'l', 'o'}, {'o', 'l', 'l', 'e', 'h'}},
        {{'H', 'a', 'n', 'n', 'a', 'h'}, {'h', 'a', 'n', 'n', 'a', 'H'}},
    };

    Solution s;

    for (auto tc : tcs)
    {
        s.reverseString(tc.s);
        CHECK(tc.s == tc.want);
    }
}
